apiclient/rancher_api/managers.py

```diff
@@ -346,6 +346,24 @@ def create(self, name, cluster_id, *, raw=False):
         return self._create(self.PATH_fmt.format(cluster_id=cluster_id), json=data, raw=raw)
 
 
+class ChartManager(BaseManager):
+    PATH_fmt = "k8s/clusters/{cluster_id}/v1/catalog.cattle.io.apps"
+    CREATE_fmt = "k8s/clusters/{cluster_id}/v1/catalog.cattle.io.clusterrepos/rancher-charts"
+
+    def get(self, cluster_id, namespace, name, raw=False):
+        url = self.PATH_fmt.format(cluster_id=cluster_id)
+        if namespace:
+            url = f"{url}/{namespace}"
+            if name:
+                url = f"{url}/{name}"
+        return self._get(url, raw=raw)
+
+    def create(self, cluster_id, namespace, name, raw=False):
+        url = self.CREATE_fmt.format(cluster_id=cluster_id) + "?action=install"
+        data = ChartSpec(cluster_id, namespace, name).to_dict()
```

Comment:
The Spec is not designed as harvester_api, it would be a bit confusing.
I would suggest that if we are not let `Spec` as class attribute have more  customization, we can simply use `create_data` instead.  (then there would not need the `ChartSpec`)

Reply:
Thanks for the comment.
Already put it into refactoring workitem. Will fix in https://github.com/harvester/tests/issues/960#issue-1941028775